File: Classes/WMISet/WMISet/Examples/delphi/WmiMethods/ScheduledJob/FrmNewJobU.pas
unit FrmNewJobU;

interface

uses
  Windows, Messages, SysUtils,
  {$IFDEF Delphi6}
  Variants,
  {$ENDIF}
  Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, CheckLst, ComCtrls, Buttons;

type
  TFrmNewJob = class(TForm)
    Label1: TLabel;
    spbCommand: TSpeedButton;
    lblStartTime: TLabel;
    lblDaysOfWeek: TLabel;
    lblDaysOfMonth: TLabel;
    edtCommand: TEdit;
    dtpStartTime: TDateTimePicker;
    chbRunRepeatedly: TCheckBox;
    clbDaysOfWeek: TCheckListBox;
    chbInteract: TCheckBox;
    clbDaysOfMonth: TCheckListBox;
    btnCreate: TButton;
    btnCancel: TButton;
    odCommand: TOpenDialog;
    procedure spbCommandClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure edtCommandChange(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
    function getDaysOfWeek: integer;
    function getDaysOfMonth: integer;
  end;

var
  FrmNewJob: TFrmNewJob;

implementation

{$R *.dfm}

procedure TFrmNewJob.spbCommandClick(Sender: TObject);
begin
  if (odCommand.Execute) then
    edtCommand.Text := odCommand.FileName;
end;


function TFrmNewJob.getDaysOfWeek: integer;
var
  vMask, i: integer;
begin
  vMask  := 1;
  Result := 0;
  for i := 0 to clbDaysOfWeek.Items.Count - 1 do
  begin
    if (clbDaysOfWeek.Checked[i]) then Result := Result or vMask;
    vMask := vMask * 2;
  end;
end;

function TFrmNewJob.getDaysOfMonth: integer;
var
  vMask, i: integer;
begin
  vMask  := 1;
  Result := 0;
  for i := 0 to clbDaysOfMonth.Items.Count - 1 do
  begin
    if (clbDaysOfMonth.Checked[i]) then Result := Result or vMask;
    vMask := vMask * 2;
  end;
end;

procedure TFrmNewJob.FormCreate(Sender: TObject);
var
  i: integer;
begin
  for i := 0 to clbDaysOfMonth.Items.Count - 1 do clbDaysOfMonth.Checked[i] := true;
  for i := 0 to clbDaysOfWeek.Items.Count - 1 do clbDaysOfWeek.Checked[i] := true;
  dtpStartTime.DateTime := Now;
end;

procedure TFrmNewJob.edtCommandChange(Sender: TObject);
begin
  btnCreate.Enabled := edtCommand.Text <> '';
end;

end.

File: Classes/WMISet/WMISet/Examples/cbuilder/WmiMethods/ScheduledJob/FrmNewJobU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FrmNewJobU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
//---------------------------------------------------------------------------
__fastcall TFrmNewJob::TFrmNewJob(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TFrmNewJob::spbCommandClick(TObject *Sender)
{
  if (odCommand->Execute()) {
    edtCommand->Text = odCommand->FileName;
  }
}
//---------------------------------------------------------------------------
int __fastcall TFrmNewJob::getDaysOfWeek() {
  int mask = 1;
  int result = 0;
  for (int i = 0; i < clbDaysOfWeek->Items->Count; i++) {
      if (clbDaysOfWeek->Checked[i]) {
          result = result | mask;
      }
      mask *= 2;
  }
  return result;
}
//---------------------------------------------------------------------------
int __fastcall TFrmNewJob::getDaysOfMonth() {
  int mask = 1;
  int result = 0;
  for (int i = 0; i < clbDaysOfMonth->Items->Count; i++) {
      if (clbDaysOfMonth->Checked[i]) {
          result = result | mask;
      }
      mask *= 2;
  }
  return result;
}
void __fastcall TFrmNewJob::FormCreate(TObject *Sender)
{
  for (int i = 0; i < clbDaysOfMonth->Items->Count; i++) {
      clbDaysOfMonth->Checked[i] = true;
  }
  for (int i = 0; i < clbDaysOfWeek->Items->Count; i++) {
      clbDaysOfWeek->Checked[i] = true;
  }
}
//---------------------------------------------------------------------------
void __fastcall TFrmNewJob::edtCommandChange(TObject *Sender)
{
  btnCreate->Enabled = edtCommand->Text != "";
}
//---------------------------------------------------------------------------
